mirrord/agent/src/entrypoint.rs

```diff
@@ -804,11 +804,19 @@ async fn start_agent(args: Args) -> AgentResult<()> {
     Ok(())
 }
 
-async fn clear_iptable_chain() -> Result<(), IPTablesError> {
-    let ipt = new_iptables();
+async fn clear_iptable_chain(ipv6: bool) -> Result<(), IPTablesError> {
+    let ipt = if ipv6 {
+        IPTablesWrapper::from(new_ip6tables())
+    } else {
+        IPTablesWrapper::from(new_iptables())
+    };
 
-    let tables = SafeIpTables::load(IPTablesWrapper::from(ipt), false).await?;
-    tables.cleanup().await?;
+    if !SafeIpTables::list_mirrord_rules(&ipt).await?.is_empty() {
+        let tables = SafeIpTables::load(ipt, false).await?;
+        tables.cleanup().await?;
+    } else {
+        trace!("No mirorrd rules found, skipping iptables cleanup.");
+    }
 
     Ok(())
```

Comment:
```suggestion
    if !SafeIpTables::list_mirrord_rules(&ipt).await?.is_empty() {
        let tables = SafeIpTables::load(ipt, false).await?;
        Ok(tables.cleanup().await?)
    } else {
        trace!("No mirorrd rules found, skipping iptables cleanup.");
        Ok(())
    }
```

nit: I would do something like this (personal preference for returning expressions), but it's up to you (don't think we ever discussed anything about this particular style thing).

Reply:
Updated, thanks!